lib/inspec/resources/groups.rb

```diff
@@ -165,21 +165,38 @@ def groups
   # This uses `dscacheutil` to get the group info instead of `etc_group`
   class DarwinGroup < GroupInfo
     def groups
-      group_info = inspec.command("dscacheutil -q group").stdout.split("\n\n")
+      group_by_id = `dscl . -list /Groups PrimaryGroupID`.lines.map { |l| name, id = l.split; [id.to_i, name] }.to_h
```

Comment:
We can't use backticks - that will run locally. InSpec is supposed to run its commands on the scanning target, that's why it is running them through inspec.command.

Unless we somehow know that we are always local here, which seems a bad assumption?

Reply:
done!